### PR TITLE
spark: omit openjdk from runtime deps

### PR DESCRIPTION
**Pull request by Dentrax**
This PR omits `openjdk-17` dependency on the runtime, to allow to use with both 11 and 17 versions.

Upstream image built with JDK8.

And since they [support](https://github.com/apache/spark/blob/6bf6088e7719eccf050910d6ae495b369afb25be/python/docs/source/getting_started/install.rst#python-versions-supported) Python versions starting from > 3.9, the related Python version will be added in the image.

### CODE REVIEW

File: spark-3.5.yaml [hunk 1]
```diff
@@ -128,8 +130,9 @@ test:
   environment:
     contents:
       packages:
-        - R
         - python3
+        - openjdk-17
```
Comment (EyeCantCU) [inline spark-3.5.yaml:134]: We don't need the full JDK for the test. Just our JVM package

File: spark-3.5.yaml [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
     provides:
       - spark=${{package.full-version}}
     runtime:
+      - R
+      # - python3: Python dependency will be added in the image, since upstream supports different Python versions starting from 3.9.
       - bash
       - busybox
-      - openjdk-17
-      - openjdk-17-default-jvm
+      # - openjdk-17: OpenJDK is required for running Spark, but will be added in the image, since upstream supports different Java versions.
```
Comment (EyeCantCU) [inline spark-3.5.yaml:16]: nit: we should remove these versus commenting them out

**Review by EyeCantCU — CHANGES_REQUESTED** [review]

File: spark-3.5.yaml (hunk 2 above)
Comment (Dentrax) [reply spark-3.5.yaml:16]: I was going to remove it but then thought that someone may bring those back in the future, so wanted to drop as a warning actually. Should we remove those anyway?

Comment (EyeCantCU) [reply spark-3.5.yaml:16]: We still wouldn't want to add back the full JDK though. I'd :axe: it

Comment (Dentrax) [reply spark-3.5.yaml]: Done. Removed all commented out packages but feel a bit uncomfortable actually, i'd be nice if we should drop a comment regarding to why we shouldn't have `python` and `openjdk-` package in the runtime deps? WDYT?

Comment (EyeCantCU) [reply spark-3.5.yaml]: That could certainly be helpful for anyone looking at this in the future. I'd suspect we'll have more java packages where this is the case

**Review by mritunjaysharma394 — APPROVED** [review]
lgtm

**Review by EyeCantCU — APPROVED** [review]

**Pull request merged**